src/multivariate/dirichlet.jl

```diff
@@ -154,18 +152,25 @@ end
 
 # sampling
 
-function _rand!(rng::AbstractRNG,
-                d::Union{Dirichlet,DirichletCanon},
-                x::AbstractVector{<:Real})
+function rand(rng::AbstractRNG, d::Union{Dirichlet,DirichletCanon})
+    x = map(αi -> rand(rng, Gamma(αi)), d.alpha)
+    return lmul!(inv(sum(x)), x)
```

Comment:
Maybe use `normalize!` so we also get its special-casing to avoid over/underflow?
```suggestion
    return normalize!(x, 1)
```
Also, if e.g. `d.alpha` is a `StaticArrays.StaticArray`, this would fail, right, since `x` would also be immutable? So maybe it make sense to allocate `x` first, `map!` into it, and then have this line?